nested.just

```diff
@@ -94,6 +99,124 @@ sign whichSafe hdPath='0':
+    fi
+    safe="{{chainGovernorSafe}}"
+  fi
+
+  echo "Using script ${script} with nested approvehash_in_anvil."
+  echo "Safe address: ${ownerSafe}"
+  echo "Getting signer address..."
+
+  if [ ! -z "$SIMULATE_WITHOUT_LEDGER" ]; then
+    signer=$(cast wallet address --private-key ${PRIVATE_KEY_OWNER})
+    echo "${PRIVATE_KEY_OWNER}"
+  else
+    signer=$(cast wallet address --ledger --mnemonic-derivation-path "m/44'/60'/{{hdPath}}'/0/0")
+  fi
+  echo "============ OVERRIDE SAFES SETTINGS Parent Safe:'${ownerSafe}' & Child Safe: '${safe}' ============"
+  echo "1. Set the threshold to 1 of the child safe."
+  cast rpc anvil_setStorageAt ${safe} 0x0000000000000000000000000000000000000000000000000000000000000004 0x0000000000000000000000000000000000000000000000000000000000000001 --rpc-url ${ANVIL_LOCALHOST_RPC}
+
+  echo "2. Set the owner count to 1."
+  cast rpc anvil_setStorageAt ${safe} 0x0000000000000000000000000000000000000000000000000000000000000003 0x0000000000000000000000000000000000000000000000000000000000000001 --rpc-url ${ANVIL_LOCALHOST_RPC}
+
+
+  # cast index address 0x0000000000000000000000000000000000000001 2 => 0xe90b7bceb6e7df5418fb78d8ee546e97c83a08bbccc01a0644d599ccd2a7c2e0 expected owner mapping: {0x1 -> 0xf39..., 0xf39 -> 0x1}
+  echo "3. Insert the address 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 as the sole owner of the safe."
+  cast rpc anvil_setStorageAt ${safe} 0xe90b7bceb6e7df5418fb78d8ee546e97c83a08bbccc01a0644d599ccd2a7c2e0 0x000000000000000000000000f39Fd6e51aad88F6F4ce6aB8827279cffFb92266 --rpc-url ${ANVIL_LOCALHOST_RPC}
+  
+  echo "4. Close the mapping of the owners to the sentinel address."
+  # cast index address 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 2 => 0xbc40fbf4394cd00f78fae9763b0c2c71b21ea442c42fdadc5b720537240ebac1
+  cast rpc anvil_setStorageAt ${safe} 0xbc40fbf4394cd00f78fae9763b0c2c71b21ea442c42fdadc5b720537240ebac1  0x0000000000000000000000000000000000000000000000000000000000000001 --rpc-url ${ANVIL_LOCALHOST_RPC}
+  echo "The sole Owner of the \"${safe}\" is: $(cast call ${safe}  "getOwners()(address[])" --rpc-url http://localhost:8545)"
+  echo "================================================"
+  
+  forge build
+  outputforge=$(forge script ${script} \
+    --rpc-url ${ANVIL_LOCALHOST_RPC} \
+    --private-key {{PRIVATE_KEY_OWNER}} \
+    --sig "signJson(string,address)" \
+    ${bundlePath} \
+    ${safe})
+
+  echo "Outputforge: $outputforge" > /tmp/outputforge.txt
+  
+  ## 1. Generate hash for the SC and FND
+  hash=$(cat /tmp/outputforge.txt | grep -A 1 "If submitting onchain, call Safe.approveHash on " | cut -d ':' -f 2 | tr -d '[:space:]')
+
+  if [[ -z "$hash" ]]; then
+      echo "Hash not found in the file."
+      exit 10
+  else
+      echo "Extracted Hash: $hash"
+  fi
+
+  ## 2. Approval of this hash in the SC and FND respectively.
+  cast send ${safe} "approveHash(bytes32)" ${hash} --rpc-url ${ANVIL_LOCALHOST_RPC} --private-key {{PRIVATE_KEY_OWNER}} 2>&1 /dev/null 
```

Comment:
Does this also support chain governor, for the 3/3 (e.g. Unichain) case? I think yes based on the code, but comments indicate otherwise, so just confirming

Reply:
this is supporting any safes. However, for the 3/3 will require a bit of rework for now since we need to approve his `approvedHash` also.